controllers/clustermodule_reconciler_test.go

```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package controllers
```

Comment:
Let's move the package-level ctx in controllers_suite_test.go into the setup func.

This way we ensure nobody is reusing it

Reply:
Btw, I'm not sure I understand how the package-level context can lead to flaky tests. It should be only canceled when the test process is killed.

Reply:
One state of @zhanggbj 's PR https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2431 had constantly failing tests in CI. the changes after the following comment fixed it.

https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2431#issuecomment-1765858951

Reply:
Interesting. If I find the time I try to reproduce. Would be just interesting to understand why. I'm not sure but could be that we should re-think the pattern of using a package-level ctx. I think we use it in a bunch of places in core CAPI (but not sure)

Reply:
It only can be reproduced in prow test, I cannot reproduce it locally, so not sure about the root cause.

Reply:
If it's okay I would hold this PR for now. I'm hesitant to make this change if we don't know why we need it and tests are stable without this PR

Reply:
It's OK for me to hold it until we reproduce it, this PR is just to make test cases more isolated considering the context staff.